pkg/aggregator/aggregator.go

```diff
@@ -28,21 +27,42 @@ import (
 )
 
 // usedDefinitionForSpec returns a map with all used definitions in the provided spec as keys and true as values.
-func usedDefinitionForSpec(sp *spec.Swagger) map[string]bool {
+func usedDefinitionForSpec(root *spec.Swagger) map[string]bool {
 	usedDefinitions := map[string]bool{}
-	walkOnAllReferences(func(ref spec.Ref) spec.Ref {
-		if refStr := ref.String(); refStr != "" && strings.HasPrefix(refStr, definitionPrefix) {
-			usedDefinitions[refStr[len(definitionPrefix):]] = true
+	alreadyVisited := map[string]bool{}
```

Comment:
Why the logic of alreadyVisited is removed from the walker? Wasn't it nicer for the walker to do the job locally?

Reply:
The logic does no belong into a recursive walker as it is orthogonal. Why should we mix things if we can keep them nicely separated?

Reply:
So this will make sure we won't visit a reference twice, right? I consider it a feature of the walker if we define the walker as visiting every reference of the spec once.

Reply:
We could create a reference following walker on-top of the normal walker, as a compromise.